src/main/java/core/basesyntax/figures/FigureSupplier.java

```diff
@@ -0,0 +1,40 @@
+package core.basesyntax.figures;
+
+import core.basesyntax.ColorSupplier;
+import java.util.Random;
+
+public class FigureSupplier {
+    private ColorSupplier colorSupplier = new ColorSupplier();
+
+    int getRandomProperties() {
+        return new Random().nextInt(20);
+    }
+
+    public Figure getRandomFigure() {
+        int figure = new Random().nextInt(5);
+        if (figure == 0) {
+            return new Square(colorSupplier.getRandomColor(),
+                    getRandomProperties());
+        } else if (figure == 1) {
+            return new Rectangle(colorSupplier.getRandomColor(),
+                    getRandomProperties(),
+                    getRandomProperties());
+        } else if (figure == 2) {
+            return new RightTriangle(colorSupplier.getRandomColor(),
+                    getRandomProperties(),
+                    getRandomProperties());
+        } else if (figure == 3) {
+            return new Circle(colorSupplier.getRandomColor(),
+                    getRandomProperties());
+        }
+        return new IsoscelesTrapezoid(colorSupplier.getRandomColor(),
+                getRandomProperties(),
+                getRandomProperties(),
+                getRandomProperties());
+    }

```

Comment:
lets use switch-case construction here. And lets create 5 private methods getRandomCircle(), getRandomSquare(), ... and call them in switch-case construction, this will increase code readability. (Remember private methods should follow after public as helper ones)